thread_safe_function_round_trip/node-api/package.json

```diff
@@ -7,6 +7,9 @@
   "dependencies": {
     "bindings": "~1.2.1"
   },
+  "engines": {
```

Comment:
Same comment as before, the real dependency is N-API version 4 which is supported by some version of 8.X as well.  In any case this is still a good step.

Reply:
@mhdawson you're right and my recent experience push me to say that the most useful value to check if a feature is supported or not is the N-API version.

The problem is that it's not possible to check if a Node.js version is greater than XXX because for example the 8.16.0 has `napi_thread_safe_function` instead the 10.0.0 not.

Reply:
I was wondering if we could specify  multiple ranges maybe

">= 8.16.0 < 9.0.0 || >=10.6.0"

but not sure if that is allowed by the syntax.

Reply:
Yes it's possible to specify multiple ranges. What is specified on the engines field has effect only if the `engine-strict` config flag is set.

`npm --engine-strict install`

Reply:
The example failed on macOs Node.js 8.16.0, can you run it in other OS ?
```
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CC(target) Release/obj.target/binding/binding.o
../binding.c:185:16: warning: type specifier missing, defaults to 'int' [-Wimplicit-int]
/*napi_value*/ NAPI_MODULE_INIT(/*napi_env env, napi_value exports*/) {
               ^
../binding.c:204:33: error: use of undeclared identifier 'env'
  assert(napi_define_properties(env, exports, 1, &start_work) == napi_ok);
                                ^
../binding.c:204:38: error: use of undeclared identifier 'exports'
  assert(napi_define_properties(env, exports, 1, &start_work) == napi_ok);
                                     ^
../binding.c:208:20: error: use of undeclared identifier 'env'
  assert(napi_wrap(env,
                   ^
../binding.c:209:20: error: use of undeclared identifier 'exports'
                   exports,
                   ^
../binding.c:216:10: error: use of undeclared identifier 'exports'
  return exports;
         ^
1 warning and 5 errors generated.
```

Reply:
Looks like no macro `NAPI_MODULE_INIT` on 8.x https://github.com/nodejs/node/blob/v8.x/src/node_api.h

Reply:
Refactor Module Initialization to NAPI_MODULE(NODE_GYP_MODULE_NAME, Init) works on 8.x.

Should we do it ? cc @gabrielschulhof

Reply:
NAPI_MODULE_INIT being missing on v8.x may be an oversight as we expect same code to work across all the LTS versions subject to the NAPI_VERSION supported.  Having said that,. not sure we'll be able to get that into the last v8.x release.  Also need @gabrielschulhof to comment on whether than even makes sense.   For now if we can refactor, maybe with an #ifdef so the example still shows the way we would want people to do it post 8.x ?

Reply:
`NAPI_MODULE` now works on all platform, `#ifdef` may not be necessary.
https://github.com/nodejs/node-addon-examples/blob/31f8ef9418ec387699afda78e71dc485c22d1192/1_hello_world/napi/hello.cc#L23